src/main/webapp/app/exercises/shared/exercise-headers/header-exercise-page-with-details.component.ts

```diff
@@ -25,45 +37,52 @@ export class HeaderExercisePageWithDetailsComponent implements OnChanges {
     @Input() public displayBackButton = true; // TODO: This can be removed once we are happy with the breadcrumb navigation
     @Input() public submissionPolicy?: SubmissionPolicy;
 
-    public exerciseStatusBadge = 'bg-success';
     public exerciseCategories: ExerciseCategory[];
-    public isExamMode = false;
     public dueDate?: dayjs.Dayjs;
+    public programmingExercise?: ProgrammingExercise;
+    public course: Course;
+    public individualComplaintDeadline?: dayjs.Dayjs;
+    public isNextDueDate: boolean[];
+    public statusBadges: string[];
+    public canComplainLaterOn: boolean;
+    public numberOfSubmissions: number;
 
     icon: IconProp;
 
     // Icons
     faArrowLeft = faArrowLeft;
     faQuestionCircle = faQuestionCircle;
 
-    /**
-     * Sets the status badge and categories of the exercise on changes
-     */
-    ngOnChanges(): void {
+    constructor(private complaintService: ComplaintService, private submissionSerrvice: SubmissionService) {}
+
+    ngOnInit(): void {
         this.exerciseCategories = this.exercise.categories || [];
 
-        if (this.exercise) {
-            this.setIcon(this.exercise.type);
-        }
+        this.setIcon(this.exercise.type);
 
-        if (this.exam) {
-            this.isExamMode = true;
-        }
+        this.programmingExercise = this.exercise.type === ExerciseType.PROGRAMMING ? (this.exercise as ProgrammingExercise) : undefined;
+        this.course = getCourseFromExercise(this.exercise)!;
 
-        if (this.exercise && !this.isExamMode) {
+        if (this.exam) {
+            this.setIsNextDueDateExamMode();
+        } else {
             this.dueDate = getExerciseDueDate(this.exercise, this.studentParticipation);
-        }
+            if (!this.individualComplaintDeadline) {
+                this.individualComplaintDeadline = this.complaintService.getIndividualComplaintDueDate(this.exercise, this.course, this.studentParticipation);
+            }
+            // The student can either still submit or there is a submission where the student did not have the chance to complain yet
+            this.canComplainLaterOn =
+                (dayjs().isBefore(this.exercise.dueDate) ||
+                    (!!this.studentParticipation?.submissionCount && this.studentParticipation?.submissionCount > 0 && !this.individualComplaintDeadline)) &&
```

Comment:
I might be wrong here, but I think the second predicate in this line can be omitted. The first predicate ensures that `submissionCount` is defined. To be specific, it checks that `submissionCount` has no falsy value. In case of a number, this can be **undefined or 0**. This would mean, that your second predicate is obsolete because the first predicate already checks that submissionCount is > 0.

Reply:
Yup